mqtt2kasa/log.py

```diff
@@ -17,42 +20,43 @@ def _log_handler_address(files=tuple()):
         )
         return None
 
+def is_running_in_docker():
+    return path.exists('/.dockerenv')
 
 def initLogger(testing=False):
     logger = getLogger()
-    logger.setLevel(logging.INFO)
-
-    format = (
-        "%(asctime)s [mqtt2kasa] %(module)12s:%(lineno)-d %(levelname)-8s %(message)s"
-    )
-    formatter = logging.Formatter(format)
+    logger.setLevel(logging.INFO)        
 
-    # Logs are normally configured here: /etc/rsyslog.d/*
-    logHandlerAddress = _log_handler_address(
-        ["/run/systemd/journal/syslog", "/var/run/syslog", "/device/log"]
-    )
+    if is_running_in_docker():
+        log_to_console()
+        getLogger().info("Running in a Docker container. Logs are sent to stdout only")
+    else:
+        format = (
+            "%(asctime)s [mqtt2kasa] %(module)12s:%(lineno)-d %(levelname)-8s %(message)s"
+        )
+        formatter = logging.Formatter(format)
 
-    if logHandlerAddress:
-        syslog = SysLogHandler(
-            address=logHandlerAddress, facility=SysLogHandler.LOG_DAEMON
+        # Logs are normally configured here: /etc/rsyslog.d/*
+        logHandlerAddress = _log_handler_address(
+            ["/run/systemd/journal/syslog", "/var/run/syslog", "/device/log"]
         )
-        syslog.setFormatter(formatter)
-        logger.addHandler(syslog)
-    else:
-        stdout_handler = logging.StreamHandler()
-        stdout_handler.setFormatter(formatter)
-        logger.addHandler(stdout_handler)
+
+        if logHandlerAddress:
+            syslog = SysLogHandler(
+                address=logHandlerAddress, facility=SysLogHandler.LOG_DAEMON
+            )
+            syslog.setFormatter(formatter)
+            logger.addHandler(syslog)
+        else:
+            log_to_console()
 
     if testing:
         log_to_console()
         set_log_level_debug()
 
 
 def log_to_console():
-    consoleHandler = logging.StreamHandler()
-    format = "%(asctime)s %(module)12s:%(lineno)-d %(levelname)-8s %(message)s"
-    formatter = logging.Formatter(format)
-    consoleHandler.setFormatter(formatter)
+    getLogger().removeHandler(consoleHandler)
```

Comment:
just to make sure there would be no duplicate logs. e.g. it's running in a container and `log_to_console: true` or `log_to_console: true` and it's running on a system that doesn't have ["/run/systemd/journal/syslog", "/var/run/syslog", "/device/log"]